packages/tanstack-start/src/ApolloProvider.tsx

```diff
@@ -50,19 +50,13 @@ const WrappedApolloProvider = WrapApolloProvider<{ router: AnyRouter }>(
           () =>
             new Promise<string>((resolve) => {
               ensureInitialized(router);
-
-              const id = crypto.randomUUID() as typeof event.id;
-
               function transportEvent(event: QueryEvent) {
-                event.id = id;
-                ssr.injectScript(
-                  () =>
-                    `__APOLLO_EVENTS__.push(${jsesc(event, {
-                      isScriptContext: true,
-                      wrap: true,
-                      json: true,
-                    })})`
-                );
+                const script = `__APOLLO_EVENTS__.push(${jsesc(event, {
+                  isScriptContext: true,
+                  wrap: true,
+                  json: true,
+                })})`;
+                ssr.injectScript(() => script);
```

Comment:
This previously had a bug, where the `injectScript` callback would be called later and at that point in time the AC core had actually written additional properties to `event`, which meant that unneccessary data was transported over the wire.
We now stringify the object immediately.